### ISSUE TITLE
Difference in duration as depicted on stage overview vs VSM page

### ISSUE THREAD

**Issue by kritika-singh3** [gocd/gocd#8681]
##### Issue Type

- Bug Report

##### Summary

Note: **Relevant for a pipeline with single stage**
The stage duration stated on stage overview (on dashboard, pipeline activity and VSM page) is different from the one stated on VSM page for a given pipeline and stage counter.

###### Basic environment details

* Go Version: `GoCD Version: 20.9.0 (12319-184c7cbcfbeabe576f9851f9bff84978bd48a716)`

##### Steps to Reproduce

<details>
<summary>Config to use</summary>

```
<pipeline name="p5">
      <materials>
        <git url="/tmp/materials/test" />
      </materials>
      <stage name="s">
        <jobs>
          <job name="j" timeout="10">
            <tasks>
              <exec command="ls" />
              <exec command="env">
                <runif status="passed" />
              </exec>
            </tasks>
          </job>
          <job name="j1" timeout="0">
            <tasks>
              <exec command="sleep">
                <arg>100</arg>
                <runif status="passed" />
              </exec>
              <exec command="la">
                <runif status="passed" />
              </exec>
            </tasks>
          </job>
        </jobs>
      </stage>
    </pipeline>
```

</details>

1. Run the pipeline. Let it complete
2. Rerun the job `j1` i.e. the failing one. Let it complete.
3. Visit the VSM page (via the link on dashboard)
4. Open up the stage overview.

##### Expected Results

The stage duration stated on stage overview should be the same as one stated in the pipeline box.

##### Actual Results

The durations are different.

##### Code snippets/Screenshots

<img width="851" alt="Screen Shot 2020-10-23 at 12 28 36 PM" src="https://user-images.githubusercontent.com/41165891/96966402-5cd25f80-152b-11eb-87a4-7c304702f929.png">

#### Any other info

This behaviour is not noticed if the entire stage is rerun.

**Comment by kritika-singh3** [comment]
The issue is present in pipelines with multiple stages as well but it might not be noticeable immediately.

**Comment by maheshp** [comment]
To understand how duration is calculated, lets assume a Stage S which has re-run and hence has counter 1 & 2.

**Duration calculation on VSM and Stage details page:**
```duration = stage(counter 2).createdTime - stage((counter 2)).lastTransitionedTime```

**Duration calculation on Stage overview:**
```duration = stage(counter 1).createdTime - stage((counter 2)).lastTransitionedTime```

I think we should change the duration on the Stage Overview to be same as on the Stage details and VSM page. 
I understand that as part of the re-run, jobs which are passed in the the previous runs are mapped to it guess that is the reason duration calculation considers the duration from the previous stage counter.
But a rerun would create a new Stage instance which has different createdTime, this is displayed on the stage details page. I think we should rely on this to calculate the stage duration.

**Comment by kritika-singh3** [comment]
Fixed in #8690. Verified on `GoCD Version: 20.9.0 (12318-b67833a26e7273f83019facdf22dfcfd290cdf4a)`
